rpcq/_spec.py

```diff
@@ -105,28 +106,30 @@ def get_handler(self, request):
         :param RPCRequest request: JSON RPC request
         :return: JSON RPC reply
         """
-        try:
-            rpc_handler = self.get_handler(request)
-        except RPCMethodError as e:
-            return rpc_error(request.id, str(e))
-
-        try:
-            # Run RPC and get result
-            args, kwargs = get_input(request.params)
-            result = rpc_handler(*args, **kwargs)
-
-            if asyncio.iscoroutine(result):
-                result = await result
-
-        except Exception as e:
-            if self.serialize_exceptions:
-                _traceback = traceback.format_exc()
-                _log.error(_traceback)
-                if self.provide_tracebacks:
-                    return rpc_error(request.id, "{}\n{}".format(str(e), _traceback))
+        with catch_warnings(record=True) as warnings:
```

Comment:
my reading of the docs is that `catch_warnings` will suppress the warning output to `stdout` does that mean we won't see them in the logs?

Reply:
I think you're correct. The brief lifetime of `warnings` also makes it hard to re-emit whatever's stored in there. I'll think about this a bit.

Reply:
The easiest thing to do seemed to be to make a local redefinition of `catch_warnings` that does what we want. I've pushed a commit with that change.

Reply:
Works for me.